docs/pages/en-us/Error-Handling.md

```diff
@@ -0,0 +1,114 @@
+3. A RESULT_SUCCESS response is placed on the messaging output queue, `worker-output-queue` by the worker framework.
+
+Some messages however cannot be handled by the Worker and these are described next.
+
+## Poisoned Messages
+A poisoned message is a message that a worker is unable to handle. The message is deemed poisonous during processing when repeated catastrophic failure of the worker occurs. Regardless of how many times the message is retried, the worker will not be able to handle the message in a graceful manor. The flow of a poisoned message is described in Figure 2 below:
+
+![Poisoned Messages](images/PoisonedMessage.png)
+*Figure 2 - Poisoned message flow*
+
+1. The Worker will receive a message from the messaging input queue, `worker-input-queue` and will start to process it.
+2. The processing of the message causes the worker to die unexpectedly.
+3. The Autoscaler service scales the worker back up again to facilitate a retry.
+4. The original message is re-delivered for retry where the `x-caf-worker-retry` message header is stamped on the message and incremented. Messages which cause a worker to crash are retried up to 10 times by default before the worker gives up trying to process them and marks them as poisoned. Note that the number of permitted retries is configurable within the `RabbitWorkerQueueConfiguration` file for a worker.
+5. Once the permitted number of retries is exceeded, the original message will be acknowledged.
+6. A `RESULT_EXCEPTION` response including the poisoned message details is placed on the messaging output queue `worker-output-queue` by the worker framework when the permitted number of retries is exceeded.
+
+## OutOfMemory / StackOverflow Errors
+Messages which cause OutOfMemory / StackOverflow errors also cause the Worker to crash. These are treated the same as poisoned messages and will be retried until successful or until the retry count exceeds the permitted number of retries.  If the permitted number of retries is exceeded the message will be placed on the `worker-output-queue` by the framework, with a task status of “RESULT_EXCEPTION”. See Figure 2 above for further details.
+
+## Transient Failures
+Transient errors arise as a result of a failure beyond the control of the Worker, e.g the database being down or unavailable.
+
+![Transient Failure](images/TransientFailure.png)
+*Figure 3 - Transient failure flow*
+
+1. The Worker will receive a message from the messaging input queue, `worker-input-queue` and will start to process it.
+2. The Worker is unable to process the message (unexpectedly due to some failure outside of its control - e.g. database down).
+3. The failure is reported as transient causing the Worker to be flagged as unhealthy.
+4. The Worker stops accepting work and messages intended for it are queued up within the messaging system. Messages will continue to be queued until the transient issue is addressed.
+5. The Worker continues to retry the message indefinitely.
```

Comment:
Only when the health-check starts passing again will the message be retried.

Reply:
Ok